### PR TITLE
Uppercase configuration string

### PR DESCRIPTION
**Pull request by ViktorHofer**
Upper-casing the first character to allow lower-case configurations in the CLI. Requires bash v4+ (2009 released).

### CODE REVIEW

**Comment by danmosemsft** [comment]
Is it possible there is a distro we (or community) build on that has older bash? If so you could test `$BASH_VERSION` maybe?

**Comment by ViktorHofer** [comment]
@wfurt do you know if we support unix distros with a bash version older than v4? Would be nice to know also for future script changes.

**Comment by wfurt** [comment]
Alpine 3.8 has bash 4.3.48 and Ubuntu 14.04 has bash 4.3.11 and RH6 has  4.1.2.
Those are generally oldest distributions I can think of.

**Comment by danmosemsft** [comment]
You could potentially add a line that errors nicely  if version is too old.

**Comment by ViktorHofer** [comment]
Ok good that I added the check as OSX uses an older version of bash 😂. I will change this to use a cross bash solution which doesn't require bash 4.

File: eng/build.sh [hunk 1]
```diff
@@ -80,7 +80,8 @@ while [[ $# > 0 ]]; do
       shift 2
       ;;
      -configuration|-c)
-      arguments="$arguments /p:ConfigurationGroup=$2 -configuration $2"
+      val="$(tr '[:lower:]' '[:upper:]' <<< ${2:0:1})${2:1}"
```
Comment (safern) [inline eng/build.sh]: out of curiosity what does: ` <<< ${2:0:1})${2:1}` do?

Comment (ViktorHofer) [reply eng/build.sh]: Uppercase the first character (0-1) and append the remaining ones (1-n).

**Pull request merged**